### DISCUSSION TITLE
How do I check if an object has an avalonia property

### DISCUSSION THREAD

**Discussion by Varmod** [AvaloniaUI/Avalonia#13805] [Q&A] [answered]
Hi, 

I'm trying to write a method that would try to set a BorderThickness on any object that is passed in.

So It has to know whether the object actually has the property or not, normally doing a inheritance check should be enough. But several classes have a BorderThickness and not all inherit from Border. So something like 
```c#
if (obj is Border border)
{
   border.BorderThickness = ... 
}
```
doesn't work in all cases.

I know I can use reflection, but I wanted to check whether there is a better option.
Thanks!

**Comment by workgroupengineering** [comment, marked as answer]
https://reference.avaloniaui.net/api/Avalonia/AvaloniaObject/6FD77884

**Reply by Varmod** [reply]
Thank you, AvaloniaObject is the class I was looking for. I think that in my case SetValue() will do.
